searchlib/src/vespa/searchlib/tensor/distance_functions.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "distance_function.h"
+
+namespace search::tensor {
+
+// TODO: Make distance functions hardware optimized.
+
+/**
+ * Calculates the square of the standard Euclidean distance.
+ */
+template <typename FloatType>
+class SquaredEuclideanDistance : public DistanceFunction {
+public:
+    double calc(const vespalib::tensor::TypedCells& lhs, const vespalib::tensor::TypedCells& rhs) const override {
+        auto lhs_vector = lhs.template typify<FloatType>();
```

Comment:
i don't see that you should need "template" here...

Reply:
You are right. It was (somehow) needed when this was done inside the templated HnswIndex class, but not here. Fixed.